src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -229,11 +243,38 @@ impl CopyToS3Uploader {
     /// Finishes any remaining in-progress upload.
     async fn flush(&mut self) -> Result<(), anyhow::Error> {
         if let Some(uploader) = self.current_file_uploader.take() {
-            uploader.finish().await?;
+            let current_file = self.current_file_path();
+            let handle =
+                mz_ore::task::spawn(|| "s3_uploader::finish", async { uploader.finish().await });
+            let CompletedUpload {
+                part_count,
+                total_bytes_uploaded,
+            } = handle.await.unwrap()?;
+            info!(
+                "finished upload: bucket {}, file {}, bytes_uploaded {}, parts_uploaded {}",
+                &self.bucket, current_file, total_bytes_uploaded, part_count
+            );
         }
         Ok(())
     }
 
+    async fn upload_buffer(&mut self) -> Result<(), anyhow::Error> {
+        assert!(!self.buf.is_empty());
+        assert!(self.current_file_uploader.is_some());
+
+        let mut uploader = self.current_file_uploader.take().unwrap();
+        let buf = std::mem::take(&mut self.buf);
+        // Clear and move the buffer contents into the task to do the upload.
+        let handle = mz_ore::task::spawn(|| "s3_uploader::buffer_chunk", async move {
+            let result = uploader.buffer_chunk(&buf).await;
+            (uploader, result)
+        });
+        let (uploader, result) = handle.await.unwrap();
```

Comment:
these should be `.await_and_assert_finished().await` or they can panic during tokio runtime shutdown

Reply:
oh I see, will update

Reply:
Er..I did not find the `.await_and_assert_finished` method anywhere

Reply:
no worries, just pushed. I was looking at public docs, later realized it's in mz_ore